### PR TITLE
[JB][SUPP-165] summary page, update backend cor, update test json to …

### PR DESCRIPTION
**Pull request by barnesjake**
…work with new mongo library in backend

### CODE REVIEW

File: test/controllers/PaymentPlanSetUpControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      doc.select(".hmrc-sign-out-nav__link").attr("href") shouldBe "http://localhost:9949/auth-login-stub/session/logout"
+      doc.select("#back").size() shouldBe 0
+
+      doc.select(".govuk-panel__title").text() shouldBe "Your payment plan is set up"
+      doc.select(".govuk-panel__body").text() shouldBe "Your payment reference is 123PA44545546"
+      val subheadings = doc.select(".govuk-heading-m").asScala.toList
+      val paragraphs = doc.select(".govuk-body").asScala.toList
+      subheadings(0).text() shouldBe "What happens next"
+      paragraphs(0).text() shouldBe "HMRC will send you a letter within 5 working days with your payment dates."
+      paragraphs(1).text() shouldBe "Your upfront payment will be taken within 10 working days. Your next payment will be taken on 28th August 2022 or the next working day."
```
Comment (AndyHWChung) [inline test/controllers/PaymentPlanSetUpControllerSpec.scala:62]: does the test data in this test include more than one payment in the payment plan? Just making sure that the sortBy and headOption is being tested for this line here.

Comment (barnesjake) [reply test/controllers/PaymentPlanSetUpControllerSpec.scala:62]: Yeah, the selected plan instalments has:
```"regularCollections" : [
                           {
                               "dueDate" : "2022-08-28",
                               "amountDue" : 55573
                           },
                           {
                               "dueDate" : "2022-09-28",
                               "amountDue" : 55573
                           }
                       ]```
So displaying August is correct

**Review by AndyHWChung — APPROVED** [review]

File: test/controllers/PaymentPlanSetUpControllerSpec.scala (hunk 1 above)
Comment (barnesjake) [reply test/controllers/PaymentPlanSetUpControllerSpec.scala:62]: I'm just running the unit tests again with the due dates in the array swapped around to check, if it passes I'll merge

Comment (barnesjake) [reply test/controllers/PaymentPlanSetUpControllerSpec.scala:62]: Changing the ordering of the array in the test data broke a test for CheckPaymentSchedule, so I've added this ordering there also, as well as leaving the test data to have ordering of September then August to hopefully catch anyone accidentally removing the ordering implicit

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**